File: src/commands/skill.rs
use anyhow::{Context, Result};
use std::path::Path;

const SKILL_MD: &str = include_str!("../../skill/claude-code/SKILL.md");
const REFERENCE_CONFIGURATION_MD: &str =
    include_str!("../../skill/claude-code/reference/configuration.md");

pub fn run_reference() -> Result<()> {
    print!("{REFERENCE_CONFIGURATION_MD}");
    Ok(())
}

pub async fn run_install(global: bool, config_file: Option<&Path>) -> Result<()> {
    let target = if global {
        crate::platform::home_dir()
            .context("could not determine home directory")?
            .join(".claude/skills/devrig")
    } else {
        // Find the config directory: use config_file's parent, or walk up from CWD
        let config_dir = if let Some(cf) = config_file {
            cf.parent().unwrap_or_else(|| Path::new(".")).to_path_buf()
        } else {
            // Walk up from CWD to find devrig.toml, use its parent
            find_config_dir()?
        };
        config_dir.join(".claude/skills/devrig")
    };

    tokio::fs::create_dir_all(&target)
        .await
        .with_context(|| format!("creating directory {}", target.display()))?;

    tokio::fs::write(target.join("SKILL.md"), SKILL_MD)
        .await
        .with_context(|| format!("writing SKILL.md to {}", target.display()))?;

    println!("Installed devrig skill to {}", target.display());
    println!();
    println!("Try asking Claude: \"What services are running and are there any errors?\"");

    Ok(())
}

fn find_config_dir() -> Result<std::path::PathBuf> {
    let mut dir = std::env::current_dir().context("getting current directory")?;
    loop {
        if dir.join("devrig.toml").exists() {
            return Ok(dir);
        }
        if !dir.pop() {
            // Fallback to CWD if no devrig.toml found
            return std::env::current_dir().context("getting current directory");
        }
    }
}

File: internal/commands/skill.go
package commands

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/spf13/cobra"
	devrig "github.com/steveyackey/devrig"
	"github.com/steveyackey/devrig/internal/config"
)

// NewSkillCmd manages the bundled Claude Code skill.
func NewSkillCmd(cfgFile *string) *cobra.Command {
	cmd := &cobra.Command{
		Use:   "skill",
		Short: "Manage the devrig Claude Code skill",
	}

	var global bool
	install := &cobra.Command{
		Use:   "install",
		Short: "Install the devrig skill for Claude Code",
		RunE: func(cmd *cobra.Command, args []string) error {
			var target string
			if global {
				home, err := os.UserHomeDir()
				if err != nil {
					return fmt.Errorf("could not determine home directory: %w", err)
				}
				target = filepath.Join(home, ".claude", "skills", "devrig")
			} else {
				dir, err := skillConfigDir(cfgFile)
				if err != nil {
					return err
				}
				target = filepath.Join(dir, ".claude", "skills", "devrig")
			}
			if err := os.MkdirAll(target, 0o755); err != nil {
				return fmt.Errorf("creating directory %s: %w", target, err)
			}
			if err := os.WriteFile(filepath.Join(target, "SKILL.md"), []byte(devrig.SkillMD), 0o644); err != nil {
				return fmt.Errorf("writing SKILL.md to %s: %w", target, err)
			}
			fmt.Printf("Installed devrig skill to %s\n\n", target)
			fmt.Println("Try asking Claude: \"What services are running and are there any errors?\"")
			return nil
		},
	}
	install.Flags().BoolVar(&global, "global", false, "Install to ~/.claude/skills instead of the project")

	reference := &cobra.Command{
		Use:   "reference",
		Short: "Print the full configuration reference",
		RunE: func(cmd *cobra.Command, args []string) error {
			fmt.Print(devrig.SkillReferenceConfigurationMD)
			return nil
		},
	}

	cmd.AddCommand(install, reference)
	return cmd
}

// skillConfigDir returns the directory containing the config file: the -f
// flag's parent if set, else the nearest ancestor holding devrig.toml,
// falling back to the current directory.
func skillConfigDir(cfgFile *string) (string, error) {
	if cfgFile != nil && *cfgFile != "" {
		abs, err := filepath.Abs(*cfgFile)
		if err != nil {
			return "", err
		}
		return filepath.Dir(abs), nil
	}
	if path, err := config.Resolve(""); err == nil {
		return filepath.Dir(path), nil
	}
	return os.Getwd()
}
